// saga-injectors/lib/checkStore.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = checkStore;

var _conformsTo = require('lodash/conformsTo');

var _conformsTo2 = _interopRequireDefault(_conformsTo);

var _isFunction = require('lodash/isFunction');

var _isFunction2 = _interopRequireDefault(_isFunction);

var _isObject = require('lodash/isObject');

var _isObject2 = _interopRequireDefault(_isObject);

var _invariant = require('invariant');

var _invariant2 = _interopRequireDefault(_invariant);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Validate the shape of redux store
 */
function checkStore(store) {
  var shape = {
    dispatch: _isFunction2.default,
    subscribe: _isFunction2.default,
    getState: _isFunction2.default,
    replaceReducer: _isFunction2.default,
    runSaga: _isFunction2.default,
    injectedReducers: _isObject2.default,
    injectedSagas: _isObject2.default
  };
  (0, _invariant2.default)((0, _conformsTo2.default)(store, shape), '(app/utils...) injectors: Expected a valid redux store');
}
module.exports = exports['default'];
